thrust/cmake/thrust-config.cmake

```diff
@@ -0,0 +1,322 @@
+  #    on EVERY target that links to Thrust::Thrust. This would be a burden and a
+  #    footgun for our users. Forgetting this would silently pull in the wrong thrust!
+  # 5) A workaround is to make a non-IMPORTED library outside of the namespace,
+  #    configure it, and then ALIAS it into the namespace (or ALIAS and then
+  #    configure, that seems to work too).
+  add_library(${ugly_name} INTERFACE)
+  add_library(${alias_name} ALIAS ${ugly_name})
+endfunction()
+
+#
+# Setup Thrust::Thrust interface target
+#
+
+# Public Thrust target that should be used by projects. It will be configured
+# to respect THRUST_HOST_SYSTEM and THRUST_DEVICE_SYSTEM.
+_thrust_declare_interface_alias(Thrust::Thrust _Thrust_Thrust)
+
+# Internal Impl target that actually holds the Thrust interface. Used by all
+# other Thrust targets.
+_thrust_declare_interface_alias(Thrust::Thrust::Impl _Thrust_Thrust_Impl)
+# Strip out the 'thrust/cmake/' from 'thrust/cmake/thrust-config.cmake':
+get_filename_component(_THRUST_INCLUDE_DIR "../.." ABSOLUTE BASE_DIR "${CMAKE_CURRENT_LIST_DIR}")
+target_include_directories(_Thrust_Thrust_Impl INTERFACE "${_THRUST_INCLUDE_DIR}")
+
+target_link_libraries(_Thrust_Thrust INTERFACE Thrust::Thrust::Impl)
+
+#
+# Define options for configuring the Thrust::Thrust target
+#
+
+# Host system option:
+set(THRUST_HOST_SYSTEM_OPTIONS CPP OMP TBB NONE)
+
+if (NOT THRUST_HOST_SYSTEM_DEFAULT)
+  set(THRUST_HOST_SYSTEM_DEFAULT CPP)
+endif()
+
+# Check if the user project specified a system to use:
+if (THRUST_HOST_SYSTEM AND NOT DEFINED CACHE{THRUST_HOST_SYSTEM})
+  # Do nothing, just use the project setting
+else()
+  # Create option:
+  set(THRUST_HOST_SYSTEM ${THRUST_HOST_SYSTEM_DEFAULT} CACHE STRING "Thrust's host system.")
+  set_property(CACHE THRUST_HOST_SYSTEM PROPERTY STRINGS ${THRUST_HOST_SYSTEM_OPTIONS})
+endif()
+
+if (NOT THRUST_HOST_SYSTEM IN_LIST THRUST_HOST_SYSTEM_OPTIONS)
+  message(FATAL_ERROR
+    "THRUST_HOST_SYSTEM=${THRUST_HOST_SYSTEM}; must be one of ${THRUST_HOST_SYSTEM_OPTIONS}")
+endif()
+
+# Device system option:
+set(THRUST_DEVICE_SYSTEM_OPTIONS CUDA CPP OMP TBB NONE)
+
+if (NOT THRUST_DEVICE_SYSTEM_DEFAULT)
+  set(THRUST_DEVICE_SYSTEM_DEFAULT CUDA)
+endif()
+
+# Check if the user project specified a system to use:
+if (THRUST_DEVICE_SYSTEM AND NOT DEFINED CACHE{THRUST_DEVICE_SYSTEM})
+  # Do nothing, just use the project setting:
+else()
+  # Create option:
+  set(THRUST_DEVICE_SYSTEM ${THRUST_DEVICE_SYSTEM_DEFAULT} CACHE STRING "Thrust's device system.")
+  set_property(CACHE THRUST_DEVICE_SYSTEM PROPERTY STRINGS ${THRUST_DEVICE_SYSTEM_OPTIONS})
+endif()
+
+if (NOT THRUST_DEVICE_SYSTEM IN_LIST THRUST_DEVICE_SYSTEM_OPTIONS)
+  message(FATAL_ERROR
+    "THRUST_DEVICE_SYSTEM=${THRUST_DEVICE_SYSTEM}; must be one of ${THRUST_DEVICE_SYSTEM_OPTIONS}")
+endif()
+
+# Undocumented options.
+# Downstream libs can set these, but we don't advertise them in the CMake GUIs:
+if (THRUST_IGNORE_DEPRECATED_CPP_DIALECT)
```

Comment:
We need to add this for `THRUST_IGNORE_DEPRECATED_CPP11` too.

Reply:
Fixed on my local branch.